pkg/services/dashboards/service/zanzana.go

```diff
@@ -207,43 +208,53 @@ func (dr *DashboardServiceImpl) checkDashboardsBatch(ctx context.Context, query
 	return result, nil
 }
 
-func (dr *DashboardServiceImpl) checkDashboardsCompile(ctx context.Context, query dashboards.FindPersistedDashboardsQuery, searchRes []dashboards.DashboardSearchProjection, remains int64) ([]dashboards.DashboardSearchProjection, error) {
+func (dr *DashboardServiceImpl) getCheckCompileFn(ctx context.Context, query dashboards.FindPersistedDashboardsQuery) checkDashboardsFn {
```

Comment:
we should return `(checkDashboardsFn, error)` so we don't have to call search if anything goes wrong here